server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -145,9 +154,18 @@ static ClusterState createDataStream(MetadataCreateIndexService metadataCreateIn
         currentState = metadataCreateIndexService.applyCreateIndexRequest(currentState, createIndexRequest, false);
         IndexMetadata firstBackingIndex = currentState.metadata().index(firstBackingIndexName);
         assert firstBackingIndex != null;
+        assert firstBackingIndex.mapping() != null : "no mapping found for backing index [" + firstBackingIndexName + "]";
+
+        String fieldName = template.getDataStreamTemplate().getTimestampField();
+        Map<String, Object> mapping = firstBackingIndex.mapping().getSourceAsMap();
```

Comment:
I wonder if it was more intuitive to require that the mapping be specified on the timestamp field in the template rather than mixing it in with the index mappings. In particular, it will be more clear that those are special, not modifiable compared to the others which can change on rollover.

Reply:
I think requiring that timestamp field mapping the be specified inside the `timestamp_field` object in a composable template makes sense. It helps highlighting that the timestamp_field and its mapping are special. 

However I do think it should be made modifiable? So that new data streams being created can use an updated field mapping? This was my motivation for keeping track of the timestamp field mapping on the `DataStream` instance, so that new backing indices would use the same valid timestamp field mapping, but when new data streams are being created from the same composable template than an updated timestamp field mapping could be used.

Reply:
++, agreed, I meant unmodifiable in the sense that template changes will not reflect on already created data-streams.

Reply:
👍 I think we can change the data_stream definition in a composable template to include the mapping for the timestamp_field mapping in a seperate change?

Reply:
Yes, that can work out too.